yellowbrick/features/base.py

```diff
@@ -215,13 +225,58 @@ def fit(self, X, y=None, **kwargs):
         """
         super(DataVisualizer, self).fit(X, y, **kwargs)
 
-        # Store the classes for the legend if they're None.
-        if self.classes_ is None:
-            # TODO: Is this the most efficient method?
-            self.classes_ = [str(label) for label in np.unique(y)]
+        self._determine_target_color_type(y)
+
+        # Compute classes and colors if target type is discrete
+        if self._target_color_type == DISCRETE:
+            # Store the classes for the legend if they're None.
+            if self.classes_ is None:
+                # TODO: Is this the most efficient method?
+                self.classes_ = [str(label) for label in np.unique(y)]
```

Comment:
Nice! Way to store the classes only on a `DISCRETE` target type! I appreciate that!